syscalls.go

```diff
@@ -27,7 +27,14 @@ var (
 // invalidBPFObjNameChar returns true if char may not appear in
 // a BPF object name.
 func invalidBPFObjNameChar(char rune) bool {
-	dotAllowed := objNameAllowsDot() == nil
+	var dotAllowed bool
+	if runtime.GOOS == "js" {
+		// In Javascript environments, BPF objects are not going to be
+		// loaded to a kernel, so we can use dots without testing.
+		dotAllowed = true
```

Comment:
Why isn't this logic in `objNameAllowsDot` instead? Also, instead of hardcoding a check on `runtime.GOOS`, we should let the syscall fail and conclude that it means a dot can be used.

Reply:
I did some reading on this last week, apparently ENOSYS is returned by much of the tinygo syscall package. This would be a good sentinel to conclude that a syscall can't be made and we need to fall back to a sane default. We can make things on the Windows side behave similarly, so high-level logic only needs to check for ENOSYS.

cc @lmb